addons/stock/models/stock_picking.py

```diff
@@ -700,6 +700,8 @@ def write(self, vals):
             after_vals['location_id'] = vals['location_id']
         if vals.get('location_dest_id'):
             after_vals['location_dest_id'] = vals['location_dest_id']
+        if self.partner_id:
```

Comment:
Use `vals` instead of `self` (otherwise, each time we write on the picking, we will write the partner of all SM)